src/PropertySuggester/GetSuggestions.php

```diff
@@ -88,12 +89,17 @@ public function execute() {
 		);
 
 		if ( $params->entity !== null ) {
-			$suggestions = $suggestionGenerator->generateSuggestionsByItem(
-				$params->entity,
-				$params->suggesterLimit,
-				$params->minProbability,
-				$params->context
-			);
+			try {
+				$suggestions = $suggestionGenerator->generateSuggestionsByItem(
+					$params->entity,
+					$params->suggesterLimit,
+					$params->minProbability,
+					$params->context
+				);
+			} catch ( InvalidArgumentException $ex ) {
+				$id = $params->entity;
+				$this->dieWithError( "Invalid id: $id", 'no-such-entity', [ 'id' => $id ] );
```

Comment:
"Invalid ID" is a bit misleading. The same exception is thrown in both cases: invalid ID, and entity not found. I would reuse the existing `$ex->getMessage()` here.